app/notification-queue/server/NotificationQueue.ts

```diff
@@ -92,6 +93,10 @@ class NotificationClass {
 	}
 
 	push({ uid, rid, mid }: INotification, item: INotificationItemPush): void {
+		if (settings.get('Push_enable') !== true) {
```

Comment:
this function is called when the queue is actually being "cleared".

the idea is to not even add to the queue, so the `Push_enable` setting should be checked earlier in the flow.

here is where items are added to the queue:
https://github.com/RocketChat/Rocket.Chat/blob/142b3171170b321f55e63f63b2224a0d6cabac45/app/lib/server/lib/sendNotificationsOnMessage.js#L181

but I think the check could be done in the `shouldNotifyMobile` function:
https://github.com/RocketChat/Rocket.Chat/blob/142b3171170b321f55e63f63b2224a0d6cabac45/app/lib/server/functions/notifications/mobile.js#L74